kv/config.go

```diff
@@ -26,6 +26,11 @@ type Config struct {
 	IdleTimeout      time.Duration `mapstructure:"idle_timeout"`
 	IdleCheckFreq    time.Duration `mapstructure:"idle_check_freq"`
 	ReadOnly         bool          `mapstructure:"read_only"`
+	TLSConfig        TLSConfig     `mapstructure:"tls"`
```

Comment:
Should be a pointer: `*TLSConfig` to distinguish between empty and non set values.